docs/DeveloperGuide.adoc

```diff
@@ -153,6 +153,108 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+// tag::loans[]
+=== Loans Feature
+==== Implementation
+
+The Loans feature exists outside of the Account/Transaction mechanisms.
+It adds a separate `LoansManager` alongside the main `AccountsManager`, with `Loan` objects stored internally in an `internalList`.
+
+The following class diagram demonstrates the association between the `LoansManager` and `Loan` objects
+(miscellaneous methods such as getters and setters are omitted):
+
+image::LoanClassDiagram.png[]
+
+Each `Loan` object has the following attributes:
+
+* `person:Person` -- The person to whom the loan belongs.
+* `amount:Amount` -- The amount of money loaned.
+* `direction:Direction` -- The direction of the loan (either *in* or *out*).
+* `status:Status` -- The status of the loan (either *paid* or *unpaid*).
+* `date:Date` -- The date of the loan.
+* `description:Description` (optional) -- A description of the loan.
+
+To facilitate the manipulation of `Loan` objects, `LoansManager` implements the following operations.
+
+* `LoansManager#getLoans()` -- Gets the current list of loans.
+* `LoansManager#addLoan(Index toAdd)` -- Adds the `Loan toAdd` to `internalList`.
+* `LoansManager#editLoan(Index toEdit, Loan editedLoan)` -- Edits the loan at `Index toEdit` to match `Loan editedLoan`.
+* `LoansManager#updateLoanStatus(Index toUpdate, Status newStatus)` -- Updates the `Status` of the loan at `Index toUpdate` to `newStatus`.
+* `LoansManager#deleteLoan(Index toDelete)` -- Deletes the loan at `Index toDelete` from `internalList`.
+
+After each of the above operations, the state of `internalList` is saved in the file `loans.json`, which is stored on the local hard disk.
+
+Given below is an example usage scenario and how the `LoansManager` behaves at each step.
```

Comment:
This was very helpful for me to understand how your application works. Great job!